pkg/cri/containerd.go

```diff
@@ -254,22 +324,17 @@ func (c *Containerd) createContainer(image string,
 
 		// Check if AWS CLI is installed
 		var cmd *exec.Cmd
-		var awsPath string
-
-		if awsPath, err = exec.LookPath("aws"); err == nil {
-			// Get ECR password
-			cmd = exec.Command(awsPath, "ecr", "get-login-password", "--region", "us-east-2")
-			ecrPasswordBytes, err := cmd.Output()
-			if err != nil {
-				// Return an error if neither file exists
-				journal.Error("Failed to pull image: Error retrieving ECR password",
-					"containerName", containerName,
-					"image", image)
-				return nil, err
-			}
-			ecrPassword := strings.TrimSpace(string(ecrPasswordBytes))
-			cmd = exec.Command(ctrPath, "-n", "k8s.io", "images", "pull", "--user", fmt.Sprintf("AWS:%s", ecrPassword), image)
-		} else {
+		var ecr_region string
+		// First check if it's an ECR image
+		ecr_region, err = extractECRRegion(image)
```

Comment:
I think that you need to diffrenciate between if you are failing to parse because regex reasons and fail to parch if the maching image is from completely different registry.
e.g if it is from ecr and you fail to parse, then return an error
if it is from gcr and you fail to parse, then just return something that wont require you to log - just to avoid false positive issues